### ISSUE TITLE
Attempting to bind specific interface using interfaces option doesn't work

### ISSUE THREAD

**Issue by kdart** [jstasiak/python-zeroconf#193]
The documentation implies that `Zeroconf(interfaces=[<localIP>])` should restrict listening to one port. But it did not work because the actual socket is bound to all ports. 

The following patch fixes it.

```diff
1747c1747
< def new_socket(port: int = _MDNS_PORT) -> socket.socket:
---
> def new_socket(port: int = _MDNS_PORT, address: str = "") -> socket.socket:
1778c1778
<     s.bind(('', port))
---
>     s.bind((address, port))
1840c1840
<                 respond_socket = new_socket()
---
>                 respond_socket = new_socket(address=i)
1843c1843
<                 respond_socket = new_socket(port=0)
---
>                 respond_socket = new_socket(port=0, address=i)
```

**Comment by garethsb-sony** [comment]
Thanks, @kdart!

@jstasiak, I've tested this patch on v0.23.0 (on Windows) and it works for me too. I started to turn it into a PR, but I'm not quite sure what to do about the interaction with the new IPv6 support on master. `socket.bind` for IPv6 sockets still needs `host` and `port` (at least), and we only have interface index currently?

**Comment by garethsb-sony** [comment]
@jstasiak Do you have any thoughts on whether this approach is good, and could be integrated with master? It would be very useful to _either_ be able to restrict browsing to one port _or_ to ensure all interfaces' advertisements are discovered and reported. At the moment it seems a bit of a lottery. Thanks.

**Comment by kdart** [comment]
Honestly, I think this is a bit of a quick fix, but could be better. I'll see if I can look at this again.

**Comment by jstasiak** [comment]
Hey @kdart and @garethsb-sony, thanks for the path and the information, I totally forgot about this – I'll try to have a closer look at it soon.

**Comment by jstasiak** [comment]
@garethsb-sony @kdart as you mentioned it was a bit more involved to integrate this with IPv6 support, but I managed to merge your suggestion with @estyrke's patch from #188 and I ended up with #270 with all tests passing. Do you mind testing the version of the code from that pull request (branch `rebase-188`) and letting me know if it addresses the problem for you?

**Comment by garethsb-sony** [comment]
Thank you, @jstasiak. I will make some time to evaluate (I need to refresh my memory on the set up I had with DNS-SD services being advertised/discovered on multiple interfaces).

**Comment by jstasiak** [comment]
I just merged #270 and released version 0.28.0, if effectively does what you propose in the original comment here, @kdart. I'll close this issue now as I consider it handled, please reopen it though (or create a new one) if you still have issues with this.
